python/langsmith/client.py

```diff
@@ -4535,11 +4535,32 @@ def create_examples(
                 )
             ]
 
-        return self._upload_examples_multipart(
-            dataset_id=cast(uuid.UUID, dataset_id),
-            uploads=uploads,
-            dangerously_allow_filesystem=dangerously_allow_filesystem,
-        )
+        try:
+            return self._upload_examples_multipart(
+                dataset_id=cast(uuid.UUID, dataset_id),
+                uploads=uploads,
+                dangerously_allow_filesystem=dangerously_allow_filesystem,
+            )
+        except Exception:
+            for upload in uploads:
+                if dump_model(upload).get("attachments", None):
+                    raise ValueError(
+                        "Must upgrade your LangSmith version to use attachments"
+                    )
```

Comment:
why do you need to do a `dump_model` here? Also if attachments aren't present just discard the attachments and log a warning